docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+```java
+// Register KafkaEnpoint as "ORDERS" table with kafka topic and data format as CSV
+sqlEnv = sqlEnv.registerTable( 
+                              "ORDERS", 
+                              new KafkaEndpoint("localhost:9090", 
+                                                "inputTopic", 
+                                                new CSVMessageFormat(<>))
+                            );
+
+// Register FileEndpoint as "SALES" table with file path and data format as csv
+sqlEnv = sqlEnv.registerTable( 
+                              "SALES", 
+                              new FileEndpoint("/tmp/output", 
+                                               "out.file", 
+                                               new CSVMessageFormat(<>))
+                            );
+```
+
+**_"registerTable"_** API takes the name of the table and an instance of endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for the data. Each of these endpoints requires a different type of configurations and possibly data formats. In this example, data format is defined using _MessageFormat_ by specifying it at the endpoint creation. _CSVMessageFormat_ can be configured with schema string as follows:
+```sql
+{
+  "separator": ",",
+  "quoteChar": "\"",
+  "fields": [
+    {
+      "name": "RowTime",
+      "type": "Date",
+      "constraints": {
+        "format": "dd/MM/yyyy hh:mm:ss Z"
+      }
+    },
+    {
+      "name": "id",
+      "type": "Integer"
+    },
+    {
+      "name": "Product",
+      "type": "String"
+    },
+    {
+      "name": "units",
+      "type": "Integer"
+    }
+  ]
+}
+```
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Following data endpoints are supported: 
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
+* **FileEndpoint**
+: It needs to be configured with directory path, file name and MessageFormat as in line 2 in above code 
+* **StreamEndpoint** 
+: This allows us to connect existing operator output or input to SQL query as a data source or sink respectively. StreamEndpoint needs immediate downstream operator's input port or immediate upstream operator's output port and field mapping for CSV data or POJO class. This will be explained in detail in next [example](#fusion-style-sql-application).
+
+
+### Using User Defined functions ( UDF ) in SQL query
+
+We can use our own scalar UDF, implemented in Java, in a SQL statement for data manipulation. But before the usage, we need to register the function with execution environment by using registerFunction API.
```

Comment:
manipulation. But before the usage, we need to => manipulation but first, we need to
with execution => with the execution
using registerFunction API. => using the `registerFunction` method.